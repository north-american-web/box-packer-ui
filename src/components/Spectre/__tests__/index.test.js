import React from 'react';
import 'jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import {Panel, Toast, Bar} from '../index'

afterEach(cleanup);

describe('<Panel/>', () => {
    it('renders and displays correctly', () => {
        const footer = <div>Footer</div>;
        const children = <div>Children</div>;

        const { getByTestId, getByRole, asFragment } = render(<Panel title="Fake title" footer={footer} children={children} />);
        expect(asFragment()).toMatchSnapshot();
        expect(getByRole('heading')).toHaveTextContent('Fake title');
        expect(getByTestId('panel-body')).toHaveTextContent('Children');
        expect(getByTestId('panel-footer')).toHaveTextContent('Footer');
    })
});

describe('<Toast/>', () => {
    it('renders and displays correctly', () => {
        const children = <div>Children</div>;
        const { getByText, getByTestId, asFragment } = render(<Toast status='fake-status' children={children} />);
        expect(getByText('Children')).not.toBeEmpty();
        expect(getByTestId('toast-element')).toHaveClass('toast-fake-status');
        expect(asFragment()).toMatchSnapshot();
    })
});

describe('<Bar/>', () => {
    it('renders and displays correctly', () => {
        const { getByTestId, asFragment } = render(<Bar indicatorClasses='test-class' percent='55' />);
        expect(getByTestId('bar-element')).toHaveAttribute('aria-valuenow', '55');
        expect(getByTestId('bar-element')).toHaveAttribute('style', 'width: 55%;');
        expect(asFragment()).toMatchSnapshot();
    })
});