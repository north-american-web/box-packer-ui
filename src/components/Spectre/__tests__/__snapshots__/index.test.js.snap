// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar/> renders and displays correctly 1`] = `
<DocumentFragment>
  <div
    class="bar bar-sm"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="55"
      class="test-class"
      data-testid="bar-element"
      role="progressbar"
      style="width: 55%;"
    />
  </div>
</DocumentFragment>
`;

exports[`<Panel/> renders and displays correctly 1`] = `
<DocumentFragment>
  <div
    class="panel mt-2"
  >
    <div
      class="panel-header"
    >
      <div
        class="panel-title"
        role="heading"
      >
        Fake title
      </div>
    </div>
    <div
      class="panel-body"
      data-testid="panel-body"
    >
      <div>
        Children
      </div>
    </div>
    <div
      class="panel-footer"
      data-testid="panel-footer"
    >
      <div>
        Footer
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Toast/> renders and displays correctly 1`] = `
<DocumentFragment>
  <div
    class="toast toast-fake-status"
    data-testid="toast-element"
  >
    <div>
      Children
    </div>
  </div>
</DocumentFragment>
`;
